src/routes/users.ts

```diff
@@ -1,48 +1,74 @@
-import express, { Request, Response } from 'express';
-import { CreateUserParams } from '../interfaces/index';
-import { getUsers, createUser, getUserById, deleteUser, updateUser } from '../controllers/users';
+import express, { NextFunction, Request, Response } from 'express';
+import {
+  CreateUserParams,
+  createUserParams,
+  getUserParams,
+  loginUserParams,
+  LoginUserParams,
+  updateUserParams,
+} from '../interfaces/index';
+import { getUsers, createUser, getUserById, deleteUser, updateUser, login } from '../controllers/users';
 
 const router = express.Router();
 
 router.get('/', async (req: Request, res: Response) => {
   try {
-    const data = await getUsers();
-    res.send(data);
+    const allUsers = await getUsers();
+    res.send(allUsers);
   } catch (error: any) {
     res.send({ error: error.message });
   }
 });
 
-router.get('/:id', async ({ body }: { body: CreateUserParams }, res: Response) => {
+router.get('/:id', async ({ params }: Request, res: Response) => {
   try {
-    const data = await getUserById(body.id);
-    res.send(data);
+    const { id } = getUserParams.parse(params);
+    const user = await getUserById(id);
+    res.send(user);
   } catch (error: any) {
     res.send({ error: error.message });
   }
 });
 
-router.post('/', async ({ body }: { body: CreateUserParams }, res: Response) => {
+// Register
+router.post('/register', async ({ body }: { body: CreateUserParams }, res: Response, next: NextFunction) => {
   try {
-    const user = await createUser(body);
-    res.status(200).send(user);
+    const parsed = createUserParams.parse(body);
+    const createdUser = await createUser(parsed);
+
+    res.status(200).send(createdUser);
   } catch (error: any) {
-    res.send({ error: error.message });
+    next(error);
   }
 });
 
-router.delete('/:id', async ({ body }: { body: CreateUserParams }, res: Response) => {
+// Login
+router.post('/login', async ({ body }: { body: LoginUserParams }, res: Response, next: NextFunction) => {
   try {
-    const data = await deleteUser(body.id);
-    res.status(200).send(data);
+    const parsed = loginUserParams.parse(body);
+    const tokenData = await login(parsed.email, parsed.password);
```

Comment:
We usually use this syntax for these kind of situations

```suggestion
    const { email, password } = loginUserParams.parse(body);
    const tokenData = await login(email, password);
```